src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -198,7 +201,8 @@ public static void WriteParameter(this CodeWriter writer, Parameter clientParame
                 writer.Append($" = default");
             }
 
-            writer.AppendRaw(",");
+            if (isTrailingComma)
```

Comment:
Usually, we always write the trailing comma and then use writer.RemoveTrailingComa to clean it up when needed.